src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,122 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEAFULT_SIZE = 16;
+    static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    int size;
+
+    private Entry<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Entry[DEAFULT_SIZE];
+        size = 0;
+    }
+
+    private static class Entry<K, V> {
+        K key;
+        V value;
+        Entry<K, V> next;
+
+        Entry(K key, V value, Entry<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
+        int index = getIndex(key);
+        Entry<K, V> current = table[index];
+
+        if (key == null) {
+            if (current == null) {
+                table[index] = new Entry<>(key, value, null);
+                size++;
+                return;
+            } else {
+                while (current != null) {
+                    if (current.key == null) {
+                        current.value = value;
+                        return;
+                    }
+                    current = current.next;
+                }
+                table[index] = new Entry<>(key, value, table[index]);
+                size++;
+                return;
+            }
+        }
+
+        while (current != null) {
+            if (key.equals(current.key)) {
+                current.value = value;
+                return;
+            }
+            current = current.next;
+        }
 
+        Entry<K, V> newEntry = new Entry<>(key, value, table[index]);
+        table[index] = newEntry;
+        size++;
+
+        if (size >= table.length * DEFAULT_LOAD_FACTOR) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        Entry<K, V> current = table[index];
+
+        if (key == null) {
+            while (current != null) {
+                if (current.key == null) {
+                    return current.value;
+                }
+                current = current.next;
+            }
+            return null;
+        }
+
+        while (current != null) {
+            if (key.equals(current.key)) {
+                return current.value;
+            }
+            current = current.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
```

Comment:
The method 'resize' should not be public since it is only used within 'MyHashMap'. Consider changing the access modifier to 'private' to adhere to the encapsulation principle.